Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +21,43 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return ampHoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (current_ < 0 )
+        return true;
+    else
+        return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    return chargeTime_;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // This is where you can update your variables
-    // Hint: There are many different ways that the totalAmpHoursUsed can be updated
-    // i.e: Taking a running average of your data values, using most recent data points, etc.
+    double timeDifference;
+    double hours;
+    chargeTime_.setHMS(0,0,0,0);
+
+    if (numberOfRecords_ > 0)
+    {
+        timeDifference = (double)time_.msecsTo(batteryData.time)/3600000;
```

Comment:
this conversion uses a magic number, eliminate those by utilising the namespace above :)

Reply:
Also, let's keep things consistent by adding a space between operators.
`timeDifference = (double)time_.msecsTo(batteryData.time) / 3600000;`